app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+          'valid' => $valid,
+        ], 200);
+    }
+
+    public function checkWithinCountry(Request $request)
+    {
+        $polygonUuid = $request->input('uuid');
+
+        if ($polygonUuid === null || $polygonUuid === '') {
+            return response()->json(['error' => 'UUID not provided'], 200);
+        }
+
+        $geometry = PolygonGeometry::where('uuid', $polygonUuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+
+        $totalArea = DB::table('polygon_geometry')
+            ->where('uuid', $polygonUuid)
+            ->selectRaw('ST_Area(geom) AS area')
+            ->first()->area;
+
+        // Find site_polygon_id and project_id using the polygonUuid
+        $sitePolygonData = SitePolygon::where('poly_id', $polygonUuid)
+            ->select('id', 'project_id')
+            ->first();
+
+        if (! $sitePolygonData) {
+            return response()->json(['error' => 'Site polygon data not found for the specified polygonUuid'], 404);
+        }
+
+        // Find the country ISO using project_id from v2projects
+        $countryIso = DB::table('v2_projects')
+            ->where('uuid', $sitePolygonData->project_id)
+            ->value('country');
+
+        if (! $countryIso) {
+            return response()->json(['error' => 'Country ISO not found for the specified project_id'], 404);
+        }
+
+        $intersectionData = DB::table('world_countries_generalized')
+          ->where('iso', $countryIso)
+          ->selectRaw('world_countries_generalized.country AS country, ST_Area(ST_Intersection(world_countries_generalized.geometry, (SELECT geom FROM polygon_geometry WHERE uuid = ?))) AS area', [$polygonUuid])
+          ->first();
+
+        $intersectionArea = $intersectionData->area;
+        $countryName = $intersectionData->country;
+
+        $insidePercentage = $intersectionArea / $totalArea * 100;
+
+        $insideThreshold = 75;
+        $insideViolation = $insidePercentage < $insideThreshold;
+        $WITHIN_COUNTRY_CRITERIA_ID = 7;
+        $insertionSuccess = $this->insertCriteriaSite($polygonUuid, $WITHIN_COUNTRY_CRITERIA_ID, ! $insideViolation);
+
+        return response()->json([
+            'country_name' => $countryName,
+            'inside_percentage' => $insidePercentage,
+            'valid' => ! $insideViolation,
+            'geometry_id' => $geometry->id,
+            'insertion_success' => $insertionSuccess,
+        ]);
+
+    }
+
+    public function getGeometryType(Request $request)
+    {
+        $uuid = $request->input('uuid');
+
+        // Fetch the geometry type based on the UUID using SQL query
+        $query = 'SELECT ST_GeometryType(geom) AS geometry_type FROM polygon_geometry WHERE uuid = ?';
+        $result = DB::selectOne($query, [$uuid]);
+
+        if ($result) {
+            $geometryType = $result->geometry_type;
+            $valid = $geometryType === 'POLYGON' ? true : false;
```

Comment:
This can be simplified: `$valid = $geometryType === 'POLYGON';`

Reply:
changed.